### ISSUE TITLE
Crash: illegal class name 'com.harman.services.AIRRuntimeCheck'

### ISSUE THREAD

**Issue by Oldes** [Gamua/Adobe-Runtime-Support#117]
## Problem Description

When testing my quite simple ANE, when used with Harmans SDK (33.0.2.246) the application crashes on start with this error:
```
2019-10-03 15:02:07.471 15225-15225/com.foo.TestAndroidPermissions D/SurfaceView: surfaceCreated 1 com.adobe.air.AIRWindowSurfaceView{f7a6700 VFED..C.. .F....ID 0,0-1080,1920}
2019-10-03 15:02:07.471 15225-15225/com.foo.TestAndroidPermissions D/SurfaceView: surfaceChanged (1080,1920) 1 com.adobe.air.AIRWindowSurfaceView{f7a6700 VFED..C.. .F....ID 0,0-1080,1920}
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504] JNI DETECTED ERROR IN APPLICATION: illegal class name 'com.harman.services.AIRRuntimeCheck'
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]     (should be of the form 'package/Class', [Lpackage/Class;' or '[[B')
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]     in call to FindClass
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]     from boolean com.adobe.air.Entrypoints.doSplashScreen(java.lang.String, java.lang.String, java.lang.String, java.lang.String, java.lang.String)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504] "main" prio=5 tid=1 Runnable
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   | group="main" sCount=0 dsCount=0 flags=0 obj=0x74657a18 self=0x71d2cc0a00
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   | sysTid=15225 nice=-10 cgrp=default sched=0/0 handle=0x71d7e749c8
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   | state=R schedstat=( 201221141 1631038 211 ) utm=16 stm=4 core=4 HZ=100
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   | stack=0x7fdeb07000-0x7fdeb09000 stackSize=8MB
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   | held mutexes= "mutator lock"(shared held)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   native: #00 pc 0000000000397614  /system/lib64/libart.so (_ZN3art15DumpNativeStackERNSt3__113basic_ostreamIcNS0_11char_traitsIcEEEEiP12BacktraceMapPKcPNS_9ArtMethodEPv+212)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   native: #01 pc 000000000045dde8  /system/lib64/libart.so (_ZNK3art6Thread9DumpStackERNSt3__113basic_ostreamIcNS1_11char_traitsIcEEEEbP12BacktraceMapb+348)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   native: #02 pc 00000000002d59e0  /system/lib64/libart.so (_ZN3art9JavaVMExt8JniAbortEPKcS2_+1040)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   native: #03 pc 00000000002d5e60  /system/lib64/libart.so (_ZN3art9JavaVMExt9JniAbortVEPKcS2_St9__va_list+116)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   native: #04 pc 0000000000108ab0  /system/lib64/libart.so (_ZN3art11ScopedCheck6AbortFEPKcz+148)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   native: #05 pc 00000000000eed68  /system/lib64/libart.so (_ZN3art8CheckJNI9FindClassEP7_JNIEnvPKc+756)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   native: #06 pc 000000000026e5d4  /data/app/com.foo.TestAndroidPermissions-rIixHgwzgL0K9vO1q6vZFg==/lib/arm64/libCore.so (???)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   native: #07 pc 0000000000250704  /data/app/com.foo.TestAndroidPermissions-rIixHgwzgL0K9vO1q6vZFg==/lib/arm64/libCore.so (Java_com_adobe_air_Entrypoints_doSplashScreen+412)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   native: #08 pc 0000000000004e08  /data/app/com.foo.TestAndroidPermissions-rIixHgwzgL0K9vO1q6vZFg==/oat/arm64/base.odex (Java_com_adobe_air_Entrypoints_doSplashScreen__Ljava_lang_String_2Ljava_lang_String_2Ljava_lang_String_2Ljava_lang_String_2Ljava_lang_String_2+264)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   at com.adobe.air.Entrypoints.doSplashScreen(Native method)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   at com.harman.services.AIRRuntimeCheck.doSplashScreen(AIRRuntimeCheck.java:171)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   at com.adobe.air.AndroidActivityWrapper.onSurfaceInitialized(AndroidActivityWrapper.java:1428)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   at com.adobe.air.AIRWindowSurfaceView.surfaceChanged(AIRWindowSurfaceView.java:792)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   at android.view.SurfaceView.updateSurface(SurfaceView.java:766)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   at android.view.SurfaceView$2.onPreDraw(SurfaceView.java:154)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   at android.view.ViewTreeObserver.dispatchOnPreDraw(ViewTreeObserver.java:1045)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2808)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1780)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7827)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   at android.view.Choreographer$CallbackRecord.run(Choreographer.java:911)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   at android.view.Choreographer.doCallbacks(Choreographer.java:723)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   at android.view.Choreographer.doFrame(Choreographer.java:658)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:897)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   at android.os.Handler.handleCallback(Handler.java:789)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   at android.os.Handler.dispatchMessage(Handler.java:98)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   at android.os.Looper.loop(Looper.java:164)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   at android.app.ActivityThread.main(ActivityThread.java:6944)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   at java.lang.reflect.Method.invoke(Native method)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504]   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: java_vm_ext.cc:504] 
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:508] Runtime aborting...
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:508] 
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516] JNI DETECTED ERROR IN APPLICATION: illegal class name 'com.harman.services.AIRRuntimeCheck'
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]     (should be of the form 'package/Class', [Lpackage/Class;' or '[[B')
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]     in call to FindClass
2019-10-03 15:02:07.487 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]     from boolean com.adobe.air.Entrypoints.doSplashScreen(java.lang.String, java.lang.String, java.lang.String, java.lang.String, java.lang.String)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516] "main" prio=5 tid=1 Runnable
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   | group="main" sCount=0 dsCount=0 flags=0 obj=0x74657a18 self=0x71d2cc0a00
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   | sysTid=15225 nice=-10 cgrp=default sched=0/0 handle=0x71d7e749c8
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   | state=R schedstat=( 201221141 1631038 211 ) utm=16 stm=4 core=4 HZ=100
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   | stack=0x7fdeb07000-0x7fdeb09000 stackSize=8MB
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   | held mutexes= "mutator lock"(shared held)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   native: #00 pc 0000000000397614  /system/lib64/libart.so (_ZN3art15DumpNativeStackERNSt3__113basic_ostreamIcNS0_11char_traitsIcEEEEiP12BacktraceMapPKcPNS_9ArtMethodEPv+212)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   native: #01 pc 000000000045dde8  /system/lib64/libart.so (_ZNK3art6Thread9DumpStackERNSt3__113basic_ostreamIcNS1_11char_traitsIcEEEEbP12BacktraceMapb+348)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   native: #02 pc 00000000002d59e0  /system/lib64/libart.so (_ZN3art9JavaVMExt8JniAbortEPKcS2_+1040)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   native: #03 pc 00000000002d5e60  /system/lib64/libart.so (_ZN3art9JavaVMExt9JniAbortVEPKcS2_St9__va_list+116)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   native: #04 pc 0000000000108ab0  /system/lib64/libart.so (_ZN3art11ScopedCheck6AbortFEPKcz+148)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   native: #05 pc 00000000000eed68  /system/lib64/libart.so (_ZN3art8CheckJNI9FindClassEP7_JNIEnvPKc+756)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   native: #06 pc 000000000026e5d4  /data/app/com.foo.TestAndroidPermissions-rIixHgwzgL0K9vO1q6vZFg==/lib/arm64/libCore.so (???)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   native: #07 pc 0000000000250704  /data/app/com.foo.TestAndroidPermissions-rIixHgwzgL0K9vO1q6vZFg==/lib/arm64/libCore.so (Java_com_adobe_air_Entrypoints_doSplashScreen+412)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   native: #08 pc 0000000000004e08  /data/app/com.foo.TestAndroidPermissions-rIixHgwzgL0K9vO1q6vZFg==/oat/arm64/base.odex (Java_com_adobe_air_Entrypoints_doSplashScreen__Ljava_lang_String_2Ljava_lang_String_2Ljava_lang_String_2Ljava_lang_String_2Ljava_lang_String_2+264)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   at com.adobe.air.Entrypoints.doSplashScreen(Native method)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   at com.harman.services.AIRRuntimeCheck.doSplashScreen(AIRRuntimeCheck.java:171)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   at com.adobe.air.AndroidActivityWrapper.onSurfaceInitialized(AndroidActivityWrapper.java:1428)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   at com.adobe.air.AIRWindowSurfaceView.surfaceChanged(AIRWindowSurfaceView.java:792)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   at android.view.SurfaceView.updateSurface(SurfaceView.java:766)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   at android.view.SurfaceView$2.onPreDraw(SurfaceView.java:154)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   at android.view.ViewTreeObserver.dispatchOnPreDraw(ViewTreeObserver.java:1045)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2808)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1780)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7827)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   at android.view.Choreographer$CallbackRecord.run(Choreographer.java:911)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   at android.view.Choreographer.doCallbacks(Choreographer.java:723)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   at android.view.Choreographer.doFrame(Choreographer.java:658)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:897)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   at android.os.Handler.handleCallback(Handler.java:789)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   at android.os.Handler.dispatchMessage(Handler.java:98)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   at android.os.Looper.loop(Looper.java:164)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   at android.app.ActivityThread.main(ActivityThread.java:6944)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   at java.lang.reflect.Method.invoke(Native method)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516]   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
2019-10-03 15:02:07.488 15225-15225/com.foo.TestAndroidPermissions A/zygote64: runtime.cc:516] 
    
    --------- beginning of crash
2019-10-03 15:02:07.489 15225-15225/com.foo.TestAndroidPermissions A/libc: Fatal signal 6 (SIGABRT), code -6 in tid 15225 (roidPermissions)
```

I even don't have to call anything from the ANE, its enough that I include the ANE with the project.

**Comment by Oldes** [comment]
Also I think my ANE was working fine with the initial Harman's SDK versions. I cannot verify it anymore, because these were time limited.

**Comment by ajwfrost** [comment]
Hi
Yes this is an odd one, it seems to work on some devices but fails on other devices. We have a fix for this one coming out shortly, we're holding it back to get a couple of extra fixes into it but we're aiming to release this next week..
thanks

**Comment by Oldes** [comment]
Thanks... I hope the fix will be available soon, because it is a show-stopper for now.

**Comment by ajwfrost** [comment]
Fixed in 33.0.2.281 which is now available at https://airsdk.harman.com/download
Could you please confirm and then close this assuming it works for you?
thanks

**Comment by Oldes** [comment]
Seems to be OK now. Thanks.
